node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala

```diff
@@ -1,171 +1,261 @@
 package coop.rchain.node.web
 
-import cats.effect.Sync
+import cats.effect.{Effect, Sync}
 import cats.syntax.all._
 import cats.~>
 import com.google.protobuf.ByteString
 import coop.rchain.casper.PrettyPrinter
 import coop.rchain.node.api.WebApi
 import coop.rchain.node.api.WebApi._
 import coop.rchain.shared.Log
-import io.circe.generic.semiauto._
-import org.http4s.{HttpRoutes, Response}
+import io.circe.generic.semiauto.deriveEncoder
+import org.http4s.InvalidMessageBodyFailure
+import org.http4s.rho.RhoRoutes
+import org.http4s.rho.swagger.SwaggerSyntax
 
 object WebApiRoutes {
 
-  def service[F[_]: Sync: Log, M[_]: Sync](
-      webApi: WebApi[M]
-  )(implicit mf: M ~> F): HttpRoutes[F] = {
-    import coop.rchain.casper.protocol.{BlockInfo, LightBlockInfo}
-    import io.circe._
-    import io.circe.generic.auto._
-    import io.circe.syntax._
-    import org.http4s.circe._
-    import org.http4s.{EntityDecoder, EntityEncoder, InvalidMessageBodyFailure, Request}
-
-    val dsl = org.http4s.dsl.Http4sDsl[F]
-    import dsl._
-
-    implicit class ResponseErrorHandler(val res: F[Response[F]]) {
-      def getErrorMsg(ex: Throwable) = if (ex.getMessage == null) ex.toString else ex.getMessage
-      def handleResponseError =
-        res
-          .handleErrorWith {
-            // The place where all API errors are handled
-            // TODO: introduce error codes
-            case err: InvalidMessageBodyFailure =>
-              // Request JSON parse errors
-              BadRequest(s"${getErrorMsg(err).take(250)}...".asJson)
-            // Errors from BlockAPI
-            case err: BlockApiException => BadRequest(getErrorMsg(err).asJson)
-            case err: Throwable         =>
-              // Logging only unanticipated errors, not related to Block API or input parsing (user errors)
-              Log[F].error("HTTP API response error", err) *> BadRequest(getErrorMsg(err).asJson)
-          }
-    }
+  def service[F[+_]: Effect: Sync: Log, M[_]: Sync](
```

Comment:
Effect extends Sync so it's redundant.
https://typelevel.org/cats-effect/typeclasses/